### DISCUSSION TITLE
Conditionally showing remote tracking branch

### DISCUSSION THREAD

**Discussion by terminalmage** [starship/starship#2805] [Q&A] [answered]
The default format string for [`git_branch`](https://starship.rs/config/#git-branch) does not include the remote info. If I add them to the `format`, e.g. as `$remote_name/$remote_branch`, then I end up with a slash in my prompt at all times.

I would like to display the remote tracking branch, only when there _is_ one. Is there a way to do so outside of writing a script and invoking it via a custom command?

**Comment by andytom** [comment, marked as answer]
Have you tried wrapping the remote branch in braces in the format string e.g
`($remote_name/$remote branch)`? That should stop the `/` from being shown if both `$remote_name` and `$remote_branch` are empty.

**Reply by terminalmage** [reply]
I did not, I hadn't seen this before and it turns out it's defined right [here](https://starship.rs/config/#conditional-format-strings) in the configuration docs and I just skipped over it, because I know I read through the section on escapable characters that comes right after it. Thanks!
